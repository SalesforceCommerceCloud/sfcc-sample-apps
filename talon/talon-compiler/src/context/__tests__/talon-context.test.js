const path = require('path');
const TalonContext = require('../talon-context');
const { DEFAULT_OPTIONS } = require('test-talon-context');
const { stripPrefix } = require('../../utils/string');

const { templateDir } = DEFAULT_OPTIONS;

describe('talon-context', () => {
    it('prevents from reading unknown property', () => {
        expect(() => {
            return new TalonContext({ templateDir }).unknown;
        }).toThrow("Invalid context property: unknown");
    });

    it('uses default config', () => {
        const context = new TalonContext({ templateDir });

        // make paths relative so that snapshot does not vary
        const prefix = path.resolve(__dirname, '../..') + '/';
        const out = Object.entries(context).reduce((acc, [key, value]) => {
            acc[key] = stripPrefix(value, prefix);
            return acc;
        }, {});

        expect(out).toMatchSnapshot();
    });

    describe('ensures', () => {
        it('base path starts with /', () => {
            expect(() => {
                return new TalonContext({ templateDir, basePath: 'invalid path' });
            }).toThrow(`Base path does not start with a "/": invalid path`);
        });

        it('outputDir is specified', () => {
            expect(() => {
                return new TalonContext({ templateDir, outputDir: '' });
            }).toThrow(`outputDir must be specified`);
        });

        ['templateDir', 'srcDir', 'indexHtml', 'routesJson', 'themeJson'].forEach(option => {
            it(`${option} is specified`, () => {
                const config = { templateDir };
                config[option] = '';

                expect(() => {
                    return new TalonContext(config);
                }).toThrow(`${option} must be specified`);
            });

            it(`${option} exists`, () => {
                const config = { templateDir };
                config[option] = '/unknown';

                expect(() => {
                    return new TalonContext(config);
                }).toThrow(`${option} does not exist: /unknown`);
            });
        });
    });
});