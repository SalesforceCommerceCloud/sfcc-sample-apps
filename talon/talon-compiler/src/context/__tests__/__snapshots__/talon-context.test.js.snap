// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`talon-context uses default config 1`] = `
Object {
  "basePath": "",
  "indexHtml": "__tests__/fixtures/src/index.html",
  "isPreview": false,
  "labelsJson": "__tests__/fixtures/src/labels.json",
  "locale": "en_US",
  "outputDir": "__tests__/fixtures/dist",
  "routesJson": "__tests__/fixtures/src/routes.json",
  "srcDir": "__tests__/fixtures/src",
  "talonConfigJson": "__tests__/fixtures/talon.config.json",
  "templateDir": "__tests__/fixtures",
  "themeJson": "__tests__/fixtures/src/theme.json",
  "viewsDir": "__tests__/fixtures/src/views",
}
`;
